src/modal.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import ReactModal from 'react-modal'
+import { isServer } from './utils'
+
+/**
+ * A modal component with a built-in close button. Uses [`react-modal`](https://github.com/reactjs/react-modal) under the hood, and can accept any props `react-modal` does.
+ *
+ * Unlike `react-modal`, this component does not require an `isOpen` prop to render. However, that prop can still be used in the case where animations are necessary- see [this issue](https://github.com/reactjs/react-modal/issues/25).
+ *
+ * @name Modal
+ * @type Function
+ * @param {Function} onClose - A handler for closing the modal. May be triggered via the close button, and outside click, or a key press.
+ * @param {Boolean} [hideCloseButton] - A flag for hiding the default close button.
+ *
+ * @example
+ *
+ * function MyView () {
+ *   const [ modalShown, setModalShown ] = useState(false)
+ *   return (
+ *     <div>
+ *       <button onClick={() => setModalShown(true)}>Show modal</button>
+ *       {
+ *          <Modal onClose={ () => setModalShown(false) }>
+ *            This is the modal content!
+ *          </Modal>
+ *       }
+ *     </div>
+ *   )
+ * }
+ */
+
+const propTypes = {
+  onClose: PropTypes.func.isRequired,
+  hideCloseButton: PropTypes.bool,
+  children: PropTypes.node,
+}
+
+const defaultProps = {
+  hideCloseButton: false,
+}
+
+function getRootElement() {
+  // Skip in SSR mode
+  if (isServer()) return
+  // Note that this expects a root element with id "root"
+  // eslint-disable-next-line no-undef
+  return window.document.querySelector('#root')
```

Comment:
Is there a reason why we couldn't use `body` here?

Reply:
I think the only case would be if there were multiple apps mounted in the body... which is not our use case 😛 I can update!

Reply:
(this approach also wouldn't work for that case)